tests/build_rpms.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+# Copied from https://github.com/htcondor/htcondor-ce
+
+set -exu
+
+# Build source and binary RPMs.
+# SRPMs will be /tmp/rpmbuild/SRPMS/*.rpm.
+# Binary RPMs will be /tmp/rpmbuild/RPMS/*/*.rpm.
+
+OS_VERSION=$1
+BUILD_ENV=$2
+
+# Create the condor user/group for subsequent chowns,
+# using the official RHEL UID/GID
+groupadd -g 64 -r condor
+useradd -r -g condor -d /var/lib/condor -s /sbin/nologin \
+        -u 64 -c "Owner of HTCondor Daemons" condor
+
+if  [[ $OS_VERSION == 7 ]]; then
+    YUM_PKG_NAME="yum-plugin-priorities"
+else
+    YUM_PKG_NAME="yum-utils"
+fi
+
+# Clean the yum cache
+yum clean all
+
+# Exponential backoff retry for missing DNF lock file
+# FileNotFoundError: [Errno 2] No such file or directory: '/var/cache/dnf/metadata_lock.pid'
+# 2020-03-17T13:43:57Z CRITICAL [Errno 2] No such file or directory: '/var/cache/dnf/metadata_lock.pid'
+set +e
+for retry in {1..3}; do
+    yum -y -d0 update  # Update the OS packages
+    [[ $? -eq 0 ]] && break
+    sleep $((5**$retry))
+done
+set -e
+
+yum install -y epel-release $YUM_PKG_NAME
+
+# Broken mirror?
+echo "exclude=mirror.beyondhosting.net" >> /etc/yum/pluginconf.d/fastestmirror.conf
+
+if [[ $OS_VERSION != 7 ]]; then
+    yum-config-manager --enable powertools
+fi
+
+
+if [[ $BUILD_ENV == osg ]]; then
+    yum install -y https://repo.opensciencegrid.org/osg/3.5/osg-3.5-el${OS_VERSION}-release-latest.rpm
+    yum-config-manager --enable osg-upcoming-development  # make sure to grab 8.9.13
+else
+    yum install -y https://research.cs.wisc.edu/htcondor/repo/8.9/el${OS_VERSION}/release/htcondor-release-8.9-1.el${OS_VERSION}.noarch.rpm
+fi
+
+# Install packages required for the build
+yum -y install \
+    rpm-build \
+    git \
+    tar \
+    gzip \
+    libtool \
+    gcc-c++ \
+    make \
+    autoconf \
+    automake \
+    condor-classads-devel \
+    globus-gss-assist-devel \
+    globus-gsi-credential-devel \
+    globus-gsi-proxy-core-devel \
+    globus-gsi-cert-utils-devel \
+    docbook-style-xsl \
+    libxslt \
+    python3 \
+    python3-rpm
+
+# Prepare the RPM environment
+mkdir -p /tmp/rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
```

Comment:
not that it hurts to create these, but i thought rpmbuild does that for you

Reply:
Kicked off a branch with this removed. Let's find out if `rpmbuild` does the right thing!

Reply:
Had to still pre-create some dirs since we copy data into them but otherwise it looks like `rpmbuild` does what we want.